docs/operator-manual/ingress.md

```diff
@@ -32,15 +32,19 @@ metadata:
   name: argocd-server-cli
   namespace: argocd
 spec:
+  # NOTE: the port must be ignored if you have strip_matching_host_port enabled on envoy
   host: argocd.example.com:443
   prefix: /
-  service: argocd-server:443
+  service: argocd-server:80
+  regex_headers:
+    Content-Type: "^application/grpc.*$"
+  grpc: true
```

Comment:
I'm not sure this is going to provide a fully functional ArgoCD API. The issue is that ArgoCD CLI uses both gRPC and Rest over HTTP based operations provided by the API server. If I understand it correctly, this mapping will redirect only the gRPC requests to the `argocd-server` which would leave some operations unmapped.

Reply:
Oh, I wasn't aware that some operations were REST-only while some others were gRPC-only. Although, wouldn't the first mapping work normally with REST? I certainly haven't noticed any issue using the CLI with these mappings.

Reply:
Sorry for the delay in the response. You are right.. I haven't noticed the other mapping which is used by the UI. I checked the dedicated REST operations and those are not really used in the CLI. No issues there.

I think that there is a problem with your regex though. ArgoCD CLI can use different content-types to inform the server if it should use HTTP2 (`application/grpc+proto`) or HTTP1.1 (`application/grpc-web+proto`). This is defined by the presence of the `--grpc-web` flag. If you set your mapping with `grpc: true` envoy will use HTTP2 for the communication that might break the ArgoCD CLI if users use the `--grpc-web` flag. Can you please validate that your mapping works by running any command with `--grpc-web` flag?

Reply:
I've tested both `argocd login <host> --grpc-web` and `argocd login <host> --sso --grpc-web` and they seem to be logging me in as expected. Running `argocd app list --grpc-web` afterwards also seem to work.